File: software/examples/go/example_output.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/io16_v2_bricklet"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"time"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your IO-16 Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	io, _ := io16_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Configure channel 7 as output low
	io.SetConfiguration(7, 'o', false)

	// Set channel 7 alternating high/low 10 times with 100 ms delay
	for i := 0; i < 10; i++ {
		time.Sleep(100 * time.Millisecond)
		io.SetSelectedValue(7, true)
		time.Sleep(100 * time.Millisecond)
		io.SetSelectedValue(7, false)
	}

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}

File: software/examples/vbnet/ExampleOutput.vb
Imports System
Imports System.Threading
Imports Tinkerforge

Module ExampleOutput
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your IO-16 Bricklet 2.0

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim io As New BrickletIO16V2(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Configure channel 7 [A7] as output low
        io.SetConfiguration(7, "o"C, False)

        ' Set channel 7 [A7] alternating high/low 10 times with 100 ms delay
        Dim i As Integer
        For i = 0 To 9
            Thread.Sleep(100)
            io.SetSelectedValue(7, True)
            Thread.Sleep(100)
            io.SetSelectedValue(7, False)
        Next i

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module
